images/cilium/Dockerfile.dockerignore

```diff
@@ -28,4 +28,5 @@
 !/pkg
 !/plugins/cilium-cni
 !/proxylib
+!/operator
```

Comment:
this is not idea in my opinion, as cilium agent now is having depepency with operator :sweat:. Maybe it's side effect of having centralized AWS config https://github.com/cilium/cilium/commit/1b1134c7fde703fa5d1d986df129fb446a882900.

Reply:
another approach is to do a little bit refactor (i.e. move related common code to pkg). Keen to hear your inputs :+1:

Reply:
> another approach is to do a little bit refactor (i.e. move related common code to pkg)

It looks like the `operator/option` package in needed in order to access to configuration of the operator:

https://github.com/cilium/cilium/blob/master/pkg/aws/endpoints/resolver.go#L32-L41

I don't have much context on this part of the codebase but to me it doesn't look trivial to refactor the code in order to get rid of that import as `Resolver` is used as a callback in a couple of places:
```sh
➜  cilium git:(master) rg endpoints.Resolver pkg/
pkg/aws/ec2/ec2.go
73:	cfg.EndpointResolver = aws.EndpointResolverFunc(endpoints.Resolver)

pkg/policy/groups/aws/aws.go
65:	cfg.EndpointResolver = aws.EndpointResolverFunc(endpoints.Resolver)
```

/cc @tklauser which may have a bit more context on this

Reply:
> this is not ideal in my opinion, as cilium agent now is having depepency with operator :sweat:. Maybe it's side effect of having centralized AWS config 1b1134c.

The agent shouldn't have a dependency on the operator (and currently doesn't AFAIK). However, some of the packages which are also consumed by the agent depend on the `operator/option` package which, is fine as it only contains the operator options. This was the reason in the first place to split out that package.

I think we should be able to solve this issue by changing from `!operator` to `!operator/option` in `Dockerfile.ignore`.